lib/portfinder.js

```diff
@@ -76,40 +76,52 @@ exports.getPort = function (options, callback) {
     options = {};
   }
 
-  options.host = options.host || '::1';
-  var hosts = [options.host, '0.0.0.0', '127.0.0.1'];
+  var hosts = ['::1', '127.0.0.1', '0.0.0.0'];
+  if (options.host) {
+    hosts.push(options.host);
+  }
 
   // remove duplicate host arg if already in hosts array
-  for (var i = 1; i < hosts.length; i++) {
-    if (options.host === hosts[i]) {
-      hosts.shift();
-      break;
-    }
-  }
+  var uniqueHosts = hosts.filter(function(host, index) {
+    return hosts.indexOf(host) == index;
+  });
```

Comment:
we can remove this, its okay if we have an array on non unique hosts - portfinder will close the server when it finds an open port for a given host - we have no way of preventing this anyway as a hostname can alias an ip, thus our goal at the end of this method will be to just ensure all the ports found match, if they don't recurse using the highest found port as the starting point. This makes this code more resilient and much shorter(which i'll point out in a sec) - it will also make testing easier - we can pass in duplicate hosts on purpose, and that's our proof that an infinite loop is not possible